packages/libs/hd-wallet/src/utils/sign.ts

```diff
@@ -0,0 +1,83 @@
+import type { IUnsignedCommand } from '@kadena/client';
+import { sign } from '@kadena/cryptography-utils';
+
+import { HDKey } from 'ed25519-keygen/hdkey';
+
+const KDA_COIN_TYPE: number = 626;
+
+/**
+ * Convert a Uint8Array to a hexadecimal string.
+ * @param {Uint8Array} uint8Array - The array to convert.
+ * @returns {string} - Returns the hexadecimal representation of the input.
+ */
+export const uint8ArrayToHex = (uint8Array: Uint8Array): string => {
+  if (uint8Array.length === 33 && uint8Array.at(0) === 0) {
+    uint8Array = uint8Array.slice(1);
+  }
+  return [...uint8Array].map((x) => x.toString(16).padStart(2, '0')).join('');
+};
+
+/**
+ * Derive a key pair using a seed and an index.
+ * @param {Uint8Array} seed - The seed for key derivation.
+ * @param {number} index - The index for key derivation.
+ * @returns {{ privateKey: string; publicKey: string }} - Returns the derived private and public keys.
+ */
+export const deriveKeyPair = (
+  seed: Uint8Array,
+  index: number,
+): { privateKey: string; publicKey: string } => {
+  const key = HDKey.fromMasterSeed(seed).derive(
+    `m/44'/${KDA_COIN_TYPE}'/0'/0/${index}`,
+    true,
+  );
```

Comment:
The derivation path should also be an optional argument with default `m/44'/${KDA_COIN_TYPE}'/${index}'/0/0`, as we want to be able to provide a custom derivation path from the wallet (to support importing any mnemonic and derivation path)